### PR TITLE
CHECK-176 rewrite as per discussion with Caio

### PR DESCRIPTION
**Pull request by DGaffney**

### CODE REVIEW

File: app/main/controller/similarity_controller.py [hunk 1]
```diff
@@ -36,53 +36,26 @@ def get_body_for_request(self):
...
+            result = es.update(
+                id=request.json["doc_id"],
+                body={"doc": body},
+                index=app.config['ELASTICSEARCH_SIMILARITY']
+            )
+        else:
+            result = es.index(
+                body=body,
+                index=app.config['ELASTICSEARCH_SIMILARITY']
+            )
```
Comment (caiosba) [inline app/main/controller/similarity_controller.py:58]: If I'm not mistaken, the logic of this block is:

```
IF doc_id is passed
  update document with that doc_id
ELSE
  create a new document with random id
```

While what we want is:

```
IF doc_id is passed
  IF document with doc_id exists
    update document with that doc_id
  ELSE
    create document with that doc_id
ELSE
  create document with random id
```

Right?

Comment (DGaffney) [reply app/main/controller/similarity_controller.py:58]: yep! Good catch. Bad me, programming after my usual hours

File: app/main/controller/image_similarity_controller.py [hunk 2]
```diff
@@ -11,30 +11,37 @@
...
 
   @api.response(200, 'image signature successfully stored in the similarity database.')
   @api.doc('Store an image signature in the similarity database')
   @api.expect(image_similarity_request, validate=True)
   def post(self):
-    image = ImageModel.from_url(request.json['url'], request.json['context'])
+    if request.json.get("doc_id"):
+        delete_record(request.json)
+    image = ImageModel.from_url(request.json['url'], request.json.get('doc_id'), request.json['context'])
     self.save(image)
```
Comment (caiosba) [inline app/main/controller/image_similarity_controller.py:45]: Let's be sure that the `DELETE` and `INSERT` run inside the same transaction.

File: app/main/controller/image_similarity_controller.py [hunk 3]
```diff
@@ -11,30 +11,37 @@
...
+  'doc_id': fields.String(required=False, description='image ID to constrain uniqueness'),
   'threshold': fields.Float(required=False, default=0.9, description='minimum score to consider, between 0.0 and 1.0 (defaults to 0.9)'),
   'context': JsonObject(required=False, default=[], description='context')
 })
 
 def _after_log(retry_state):
   app.logger.debug("Retrying image similarity...")
 
+def delete_record(params):
+  deleted = db.session.query(ImageModel).filter(ImageModel.doc_id==params['doc_id']).delete()
```
Comment (caiosba) [inline app/main/controller/image_similarity_controller.py:24]: Let's please be sure that `params['doc_id']` is not `None`.

Comment (caiosba) [reply app/main/controller/image_similarity_controller.py:24]: Otherwise we can delete a bunch of images :)

Comment (DGaffney) [reply app/main/controller/image_similarity_controller.py]: Fair! Updated.

**Pull request merged**